Jenkinsfile

```diff
@@ -848,10 +862,27 @@ def mageTargetWin(String context, String directory, String target) {
   }
 }
 
-def withBeatsEnv(boolean archive, Closure body) {
+def getModulePattern(String toCompare) {
```

Comment:
Because this method seems processing the directory internally, I'd not delegate its call to the client code (the caller), instead calling it within the `withBeatsEnv` method.

As an example:

```groovy
//...
    withBeatsEnv(archive: true, withModule: withModule, directory: directory) {
//...
    withBeatsEnvWin(withModule: withModule, directory: directory) {
//...
if (withModule) {
    modulePattern = args.containsKey('directory') ? getModulePattern(args.directory) : error('withBeatsEnv: directory parameter is required.')
  }
//...
```

Reply:
Uhm, I'm not sure if I understood correctly, there is only one minor discrepancy with the `directory`. `withBeatsEnv` doesn't need the directory argument, and `makeTarget` does not have any directory at all but uses `withBeatsEnv`. I think we cannot move in that direction